app/lib/report_helper.rb

```diff
@@ -37,150 +37,147 @@ def view_level_name2int(view_level)
     # to be extended for all taxonomic ranks when needed
   end
 
-  def compute_taxon_zscores(report)
-    summary = TaxonSummary.connection.select_all("select * from taxon_summaries where background_id = #{report.background.id}").to_hash
-    total_reads = report.pipeline_output.total_reads
+  def select_taxons(report)
     pipeline_output_id = report.pipeline_output.id
-    data = TaxonCount.connection.select_all("select tax_id, pipeline_output_id, tax_level, count, count_type from taxon_counts where pipeline_output_id = #{pipeline_output_id}").to_hash
+    total_reads = report.pipeline_output.total_reads
+    background_id = report.background.id
+    # Note: stdev is never 0
+    TaxonCount.select("
+      taxon_counts.tax_id         AS  tax_id,
+      taxon_counts.count_type     AS  count_type,
+      taxon_counts.tax_level      AS  tax_level,
+      taxon_lineages.genus_taxid  AS  genus_taxid,
+      taxon_names.name            AS  name,
+      tgn.name                    AS  genus_name,
+      tcn.name                    AS  category_name,
+      taxon_counts.count          AS  count,
+      (count / #{total_reads}.0
+        * 1000000.0)              AS  rpm,
+      IF(
+        stdev IS NOT NULL,
+        GREATEST(-99, LEAST(99, (((count / #{total_reads}.0 * 1000000.0) - mean) / stdev))),
```

Comment:
add TODO for zscore truncation

Reply:
@eggerdesigns what is the doc link where you track zscore truncation logic?  I'd like to add a comment cross-referencing that from this code location